fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -611,9 +628,13 @@ struct DDQueueData {
 			if( foundActiveFetching || foundActiveRelocation ) {
 				rd.wantsNewServers |= rrs.wantsNewServers;
 				rd.startTime = std::min( rd.startTime, rrs.startTime );
-				if ((rrs.priority >= PRIORITY_TEAM_UNHEALTHY || rrs.priority == PRIORITY_TEAM_REDUNDANT) &&
-				    rd.changesBoundaries())
-					rd.priority = std::max( rd.priority, rrs.priority );
+				if(!hasHealthPriority) {
+					rd.healthPriority = std::max(rd.healthPriority, rrs.healthPriority);
+				}
+				if(!hasBoundaryPriority) {
```

Comment:
Similar as above, I think the `!` should be removed from the `if` statement. Am I missing something?